File: app/schemas/ai.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from uuid import UUID
from app.models.ai_parsed_input import SourceType
from app.models.appointment import AppointmentCategory


class AIParseRequest(BaseModel):
    source_type: SourceType
    raw_input: Optional[str] = None
    family_id: UUID


class ParsedAppointment(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    category: Optional[AppointmentCategory] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    recurrence_rule: Optional[str] = None
    child_name: Optional[str] = None
    confidence_score: float = 0.0


class AIParseResponse(BaseModel):
    parse_id: UUID
    source_type: SourceType
    parsed_appointments: List[ParsedAppointment]
    raw_input: Optional[str] = None
    overall_confidence: float
    suggestions: List[str] = []


class AIScheduleRequest(BaseModel):
    family_id: UUID
    proposed_appointment: ParsedAppointment


class AIScheduleResponse(BaseModel):
    has_conflicts: bool
    conflicts: List[dict] = []
    suggestions: List[str] = []
    optimal_times: List[datetime] = []


class VoiceInputRequest(BaseModel):
    family_id: UUID
    audio_url: Optional[str] = None
    transcription: Optional[str] = None


class EmailParseRequest(BaseModel):
    family_id: UUID
    email_subject: str
    email_body: str
    sender: Optional[str] = None


class ImageParseRequest(BaseModel):
    family_id: UUID
    image_url: str


class WeeklySummary(BaseModel):
    family_id: UUID
    week_start: datetime
    week_end: datetime
    total_appointments: int
    appointments_by_category: dict
    appointments_by_child: dict
    busiest_day: str
    busiest_day_count: int
    conflicts_detected: int
    suggestions: List[str] = []

File: frontend/src/api/ai.ts
import client from './client';

export interface ParsedAppointment {
  title?: string;
  description?: string;
  category?: string;
  start_time?: string;
  end_time?: string;
  location?: string;
  recurrence_rule?: string;
  child_name?: string;
  confidence_score: number;
}

export interface AIParseResponse {
  parse_id: string;
  source_type: string;
  parsed_appointments: ParsedAppointment[];
  raw_input?: string;
  overall_confidence: number;
  suggestions: string[];
}

export interface WeeklySummary {
  family_id: string;
  week_start: string;
  week_end: string;
  total_appointments: number;
  appointments_by_category: Record<string, number>;
  appointments_by_child: Record<string, number>;
  busiest_day: string;
  busiest_day_count: number;
  conflicts_detected: number;
  suggestions: string[];
}

export const aiApi = {
  parseEmail: async (familyId: string, subject: string, body: string, sender?: string): Promise<AIParseResponse> => {
    const response = await client.post('/ai/parse-email', {
      family_id: familyId,
      email_subject: subject,
      email_body: body,
      sender,
    });
    return response.data;
  },

  parseText: async (familyId: string, text: string): Promise<AIParseResponse> => {
    const response = await client.post('/ai/parse-text', {
      family_id: familyId,
      source_type: 'text',
      raw_input: text,
    });
    return response.data;
  },

  parseImage: async (familyId: string, file: File): Promise<AIParseResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await client.post(`/ai/parse-image?family_id=${familyId}`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },

  parseVoice: async (familyId: string, file: File): Promise<AIParseResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await client.post(`/ai/parse-voice?family_id=${familyId}`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },

  getWeeklySummary: async (familyId: string): Promise<WeeklySummary> => {
    const response = await client.get(`/ai/weekly-summary/${familyId}`);
    return response.data;
  },
};
